### PR TITLE
Leverage diff manifest endpoint

### PR DESCRIPTION
**Pull request by dthomson25**

### CODE REVIEW

File: src/app/applications/components/application-resource-diff/application-resource-diff.tsx [hunk 1]
```diff
@@ -16,12 +17,17 @@ export class ApplicationResourceDiff extends React.Component<ApplicationComponen
...
 
     public render() {
-        let liveState = this.props.liveState || {};
-        if (this.state.hideDefaultedFields) {
-            liveState = this.removeDefaultedFields(this.props.targetState || {}, liveState);
+        let diff = JSON.parse(this.props.diff.diffs);
+        const liveState = this.props.liveState;
+        const diffPatcher = new DiffPatcher();
+        if (!this.state.hideDefaultedFields ) {
+            if (liveState !== null && liveState.hasOwnProperty('metadata') && liveState.metadata.hasOwnProperty('annotations')) {
```
Comment (dthomson25) [inline src/app/applications/components/application-resource-diff/application-resource-diff.tsx:24]: I feel like there is a better way of doing this. @alexmt can you confirm that?

Comment (alexmt) [reply src/app/applications/components/application-resource-diff/application-resource-diff.tsx:24]: You can simplify it a little bit:

```
if (liveState && liveState.metadata && liveState.metadata.annotations) {
```

**Review by alexmt — APPROVED** [review]

File: src/app/shared/services/applications-service.ts [hunk 2]
```diff
@@ -30,6 +31,10 @@ export class ApplicationsService {
         return requests.get(`/applications/${name}/manifests`).query({name, revision}).then((res) => res.body as models.ManifestResponse);
     }
 
+    public diffManifests(appName: string, resourceName?: string, APIVersion?: string, kind?: string): Promise<models.ManifestDiffResponse> {
+        return requests.post(`/applications/${appName}/diff`).send({resourceName, APIVersion, kind}).then((res) => res.body as models.ManifestDiffResponse);
```
Comment (alexmt) [inline src/app/shared/services/applications-service.ts:35]: API returns list of strings not ManifestDiffResponse . Please use `JSON.parse` to deserialize strings into proper response.

**Review by alexmt — CHANGES_REQUESTED** [review]

File: src/app/applications/components/application-resource-diff/application-resource-diff.tsx (hunk 1 above)
Comment (dthomson25) [reply src/app/applications/components/application-resource-diff/application-resource-diff.tsx:24]: fixed

**Comment by alexmt** [comment]
Implemented as part of https://github.com/argoproj/argo-cd-ui/commit/649152c97a9fc93292584c047106952ec27f3add

**Pull request closed without merging**